src/v2/util/BaseLoginRouter.js

```diff
@@ -130,12 +134,32 @@ function (Okta, BrowserFeatures, Settings,
         this.appState.set('remediationSuccess', trans);
       });
 
+      this.listenTo(this.appState, 'change:remediationSuccess', function (appState, trans) {
+        this.handleRemediationSuccess(trans);
+      });
+
       this.listenTo(this.appState, 'navigate', function (url) {
         this.navigate(url, { trigger: true });
       });
 
     },
 
+    handleRemediationSuccess: function (trans) {
+      // transform response
+      const ionResponse = transform(trans);
+      this.appState.set('currentState', ionResponse.currentState);
+      // set formSchema
+      const formSchema = ionResponse.currentState.remediation[0].value;
+      this.appState.set('formSchema', formSchema);
+      // set setFormSchemaInputMap
+      RemediationUtil.setFormSchemaInputMap(formSchema);
```

Comment:
sounds like another derived props and `get**` function from appState instead of maintaining state in RemediationUtil